### ISSUE TITLE
Start incorporating additional data in spatialplanr vignette from oceandatr

### ISSUE THREAD

**Issue by jaseeverett** [MathMarEcol/spatialplanr#82]
Start to write some code to extract additional features from offshoredatr for the East Australia planning region. 

https://github.com/emlab-ucsb/offshoredatr

Below are some examples I wrote which use raster. Jason flower has now updated the code to accept sf objects so you should be a able to replace `planning_grid = planning_grid` with `PUs` and extract the same data.

```
bathymetry <- offshoredatr::get_bathymetry(planning_grid = planning_grid, keep = FALSE)
geomorphology <- offshoredatr::get_geomorphology(area_polygon = micro_eez, planning_grid = planning_grid)
knolls <- offshoredatr::get_knolls(area_polygon = micro_eez, planning_grid = planning_grid)
seamounts <- offshoredatr::get_seamounts_buffered(area_polygon = micro_eez, planning_grid = planning_grid, buffer_km = 30)
coral_habitat <- offshoredatr::get_coral_habitat(area_polygon = micro_eez, planning_grid = planning_grid)
enviro_regions <- offshoredatr::get_enviro_regions(area_polygon = micro_eez, planning_grid = planning_grid, max_num_clusters = 5)
```

When you are comfortable that these example work, include them in the spatialplanr.Rmd vignette BEFORE we start the prioritisation (just after the AquaMaps feature analysis and data prep)

**Comment by jaseeverett** [comment]
`offshoredatr` has now been renamed to `oceandatr` and some of the code has been simplified.

The simplified code looks like:

```
bathymetry <- oceandatr::get_bathymetry(planning_grid = planning_grid, keep = FALSE)
geomorphology <- oceandatr::get_geomorphology(planning_grid = planning_grid)
knolls <- oceandatr::get_knolls(planning_grid = planning_grid)
seamounts <- oceandatr::get_seamounts_buffered(planning_grid = planning_grid, buffer = 30000)
coral_habitat <- oceandatr::get_coral_habitat(planning_grid = planning_grid)
enviro_regions <- oceandatr::get_enviro_regions(planning_grid = planning_grid, max_num_clusters = 5)
```

**Comment by KilianGRIHAULTBARREIRO** [comment]
``` r
Region <- "Coral Sea" # "Australia"
Type <- "Oceans" # "EEZ"

Shape <- "Hexagon" # "Shape of PUs
PU_size <- 10000 # km2

cCRS <-  "+proj=moll +lon_0=0 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m no_defs"

Bndry <- splnr_get_boundary(Region, Type, cCRS)
#> Error in splnr_get_boundary(Region, Type, cCRS): impossible de trouver la fonction "splnr_get_boundary"

landmass <- rnaturalearth::ne_countries(
  scale = "medium",
  returnclass = "sf"
) %>%
  sf::st_transform(cCRS)
#> Error in rnaturalearth::ne_countries(scale = "medium", returnclass = "sf") %>% : impossible de trouver la fonction "%>%"

PUs <- splnr_get_planningUnits(Bndry, landmass, PU_size, Shape)
#> Error in splnr_get_planningUnits(Bndry, landmass, PU_size, Shape): impossible de trouver la fonction "splnr_get_planningUnits"



gfw_data <- get_gfwData('Australie', "2022-01-01", "2022-12-31", "yearly", "low", key, cCRS)
#> Error in get_gfwData("Australie", "2022-01-01", "2022-12-31", "yearly", : impossible de trouver la fonction "get_gfwData"
```

<sup>Created on 2024-01-31 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by KilianGRIHAULTBARREIRO** [comment]
For now my function is working no matter what the data format is : 
I am modifying it to handle better the color palette with the different conditions :

``` r
splnr_plot_cost_3 <- function(cost, cost_names = c("Cost1", "Cost2"), 
                              legendTitle = "Cost", 
                              paletteName = "YlGnBu",
                              plotTitle = "") {
  
  gg <- ggplot2::ggplot() 

  # Determine breaks based on the maximum absolute value across all cost_names
  max_abs_values <- sapply(cost_names, function(cost_name) max(abs(na.omit(cost[[cost_name]]))))
  if(all(sapply(cost_names, function(cost_name) all(na.omit(cost[[cost_name]]) < 0)))) {
    breaks <- seq(-max(max_abs_values), 0, length.out = 5)
  } else {
    breaks <- seq(-max(max_abs_values), max(max_abs_values), length.out = 5)
  }
  
  # Create one instance of ggplot outside the loop
  gg <- gg +
    ggplot2::coord_sf(xlim = sf::st_bbox(cost)$xlim, ylim = sf::st_bbox(cost)$ylim) +
    ggplot2::labs(subtitle = plotTitle, fill = legendTitle) 
  
  for (cost_name in cost_names) {
    gg <- gg +
      ggplot2::geom_sf(data = cost, ggplot2::aes(fill = !!rlang::sym(cost_name)),
              colour = "grey80", size = 0.1, show.legend = TRUE) +
      ggplot2::scale_fill_distiller(palette = paletteName, 
                           aesthetics = c("colour", "fill"),
                           breaks = breaks,
                           limits = c(-max(max_abs_values), max(max_abs_values)),
                           oob = scales::squish) 
  }
  
  return(gg)
}
```

<sup>Created on 2024-02-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by jaseeverett** [comment]
Thanks Kilian. Looks good. 
We also want to make it work for other data (like Bathymetry so rename the `cost` and `cost_names` to something like `data` and `col_names`

**Comment by jaseeverett** [comment]
FYI - You should really raise a new issue for this. It is not related to putting oceandatr into the vignette.

**Comment by KilianGRIHAULTBARREIRO** [comment]
Ok done! Should I remove the code I've written here ?

**Comment by jaseeverett** [comment]
Another update to the oceandatr package.

The analysis/gridding code from `oceandatr` has been moved into a new package called `spatialgridr`. So `oceandatr` contains all the data. `spatialgridr` contains the code. So the data_to_planning_grid code will now be housed in `spatialgridr`. Both packages are still available on the emlab GitHub organisation.

Also note some name changes:

From Jason Flower: 
```
I've made a lot of changes to [spatialgridr](https://github.com/emlab-ucsb/spatialgridr) and wanted to summarize these as they will have cascading effects for you both - sorry, but hopefully this will be the last name changes for a while! The renaming was partly based on a suggestion from Jase:
Function data_to_planning_grid renamed get_data_in_grid and argument planning_grid =  changed to spatial_grid = 
Function get_planning_grid renamed to get_grid and will now be housed in the spatialgridr  package
Internal functions raster_to_planning_grid and sf_to_planning_grid both dropped the planning part of their name and renamed argument planning_grid =   to spatial_grid = 
I've also built the spatialgridr [webpage](https://emlab-ucsb.github.io/spatialgridr/index.html)!
```
